### PR TITLE
Remove support for STLPort and deprecate support for gnustl on Android

### PR DESCRIPTION
**Pull request by var-const**

### CODE REVIEW

File: app/src/include/firebase/app.h [hunk 1]
```diff
@@ -55,11 +55,6 @@ namespace internal {
 class AppInternal;
 }  // namespace internal
 
-#ifdef _STLPORT_VERSION
```
Comment (var-const) [inline app/src/include/firebase/app.h:58]: Do we want a similar warning for gnustl at this point? Or would that be potentially too noisy?

**Review by var-const — COMMENTED** [review]
Please let me know if I missed anything important in the build system.

**Comment by var-const** [comment]
Looks like all the CI actions succeeded except for [macos-latest-Release-x64-static](https://github.com/firebase/firebase-cpp-sdk/pull/398/checks?check_run_id=2453211276):
```
53/103 Test  #53: firebase_auth_user_test ...........................................***Failed    3.86 sec
        Start  53: firebase_auth_user_test
      Test  #53: firebase_auth_user_test ...........................................***Failed    3.49 sec
        Start  53: firebase_auth_user_test
      Test  #53: firebase_auth_user_test ...........................................***Failed    4.09 sec
```

I presume this is not related to this PR. Is it a known issue?

File: app/src/include/firebase/app.h (hunk 1 above)
Comment (jonsimantov) [reply app/src/include/firebase/app.h:58]: I think we should have the warning. We've had deprecation warnings sit there for months before.

Comment (var-const) [reply app/src/include/firebase/app.h:58]: Done.

**Review by jonsimantov — APPROVED** [review]

**Comment by jonsimantov** [comment]
> Looks like all the CI actions succeeded except for [Desktop Builds / macos-latest-Release-x64-static](https://github.com/firebase/firebase-cpp-sdk/pull/398/checks?check_run_id=2453211276):
> 
> ```
> 53/103 Test  #53: firebase_auth_user_test ...........................................***Failed    3.86 sec
>         Start  53: firebase_auth_user_test
>       Test  #53: firebase_auth_user_test ...........................................***Failed    3.49 sec
>         Start  53: firebase_auth_user_test
>       Test  #53: firebase_auth_user_test ...........................................***Failed    4.09 sec
> ```
> 
> I presume this is not related to this PR. Is it a known issue?

I assume it's an unrelated issue or a flake, I haven't been seeing it on my PRs. This PR shouldn't affect Desktop builds at all.

**Comment by var-const** [comment]
@jonsimantov Jon, can you please re-approve? Looks like GitHub marked your review as stale after I merged `main`.

**Review by jonsimantov — APPROVED** [review]

**Pull request merged**